src/hotspot/cpu/s390/macroAssembler_s390.cpp

```diff
@@ -3152,6 +3153,301 @@ void MacroAssembler::check_klass_subtype(Register sub_klass,
   BLOCK_COMMENT("} check_klass_subtype");
 }
 
+// scans r_count pointer sized words at [r_addr] for occurrence of r_value,
+// generic (r_count must be >0)
+// iff found: CC eq, r_result == 0
+void MacroAssembler::repne_scan(Register r_addr, Register r_value, Register r_count, Register r_result) {
+  NearLabel L_loop, L_exit;
+
+  BLOCK_COMMENT("repne_scan {");
+#ifdef ASSERT
+  z_chi(r_count, 0);
+  asm_assert(bcondHigh, "count must be positive", 11);
+#endif
+
+  clear_reg(r_result, true /* whole_reg */, false /* set_cc */);  // sets r_result=0, let's hope that search will be successful
+
+  bind(L_loop);
+  z_cg(r_value, Address(r_addr));
+  z_bre(L_exit); // branch on success
+  z_la(r_addr, wordSize, r_addr);
+  z_brct(r_count, L_loop);
+
+  // z_brct above doesn't change CC.
+  // If the search operation is unsuccessful, then it's a failure case.
```

Comment:
```suggestion
  // If we reach here, then the value in r_value is not present. Set r_result to 1.
```
"Failure" is not a good word to use here, because it's unclear what failed.